sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/FileSourceStrategy.scala

```diff
@@ -230,6 +230,14 @@ object FileSourceStrategy extends Strategy with PredicateHelper with Logging {
         case MetadataStructColumn(attr) => attr
       }
 
+      // We divide metadata columns into two categories: constant and generated.
+      // For constant metadata columns, we create these attributes as non-nullable
+      //  when passing to readers, since the values remain persistent in the files.
```

Comment:
```suggestion
      //  when passing to readers, since the values are always available.
```
The values are actually not always stored, for example the path of a file is not stored in the file